x-pack/plugins/apm/public/components/app/TransactionOverview/index.tsx

```diff
@@ -56,6 +58,12 @@ export function TransactionOverviewView({
 }: TransactionOverviewProps) {
   const { serviceName, transactionType } = urlParams;
 
+  // TODO: improve urlParams typings.
+  // `serviceName` will never be undefined here, and this check should not be needed
+  if (!serviceName) {
+    return null;
+  }
```

Comment:
safe to remove this?

Reply:
Unfortunately not:
<img width="741" alt="Screenshot 2019-04-23 at 23 15 09" src="https://user-images.githubusercontent.com/209966/56616437-afbd1500-661d-11e9-9bec-4417ee430910.png">

Since `serviceName` is optional we have to make the check because `serviceName` is required in other calls below.

Reply:
Would be good if the `urlParams` were "aware" where in the application they were used. `serviceName` is not always guaranteed to be in the url - but if we are able to load this component it must be there.